mods/default/functions.lua

```diff
@@ -325,47 +325,64 @@ default.after_place_leaves = function(pos, placer, itemstack, pointed_thing)
 	end
 end
 
--- Leafdecay ABM
-
-minetest.register_abm({
-	label = "Leaf decay",
-	nodenames = {"group:leafdecay"},
-	neighbors = {"air"},
-	interval = 2,
-	chance = 10,
-	catch_up = false,
-
-	action = function(pos, node, _, _)
-		-- Check if leaf is placed
-		if node.param2 ~= 0 then
-			return
+-- Leafdecay
+local function leafdecay_after_destruct(pos, oldnode, def)
+	for _, v in pairs(minetest.find_nodes_in_area(vector.subtract(pos, def.radius),
+			vector.add(pos, def.radius), def.leaves)) do
+		local node = minetest.get_node(v)
+		if node.param2 ~= 1 then
```

Comment:
actually, manually placed leaves have param2 =1 (through place_param2) which stops leaf decay. Placing with `shift` causes them to decay normally, since that code puts param2 at `0`.

Here we start the leafdecay timer for nodes that need to decay, so `== 0` is good, and so is `~= 1`.

I'll change it to `== 0` though, since that's a bit simpler to understand.